### PR TITLE
[0-size Tensor No.180、182-183] Add 0-size Tensor support for hinge_embedding_loss

### PR DESCRIPTION
**Pull request by co63oc**
### PR Category

Execute Infrastructure

### PR Types

Improvements 

### Description

180 paddle.nn.functional.hinge_embedding_loss 
GPU 测试通过
CPU 测试通过
182 paddle.nn.functional.kl_div 
GPU 测试通过
image
CPU 测试通过
image

183 paddle.nn.functional.l1_loss
GPU 测试通过
CPU 测试通过

paddle.nn.functional.kl_div 
kernel 修改 impl和xpu调用
前向reduction不为none时，返回填充NAN
infermeta没有修改

### CODE REVIEW

**Comment by co63oc** [comment]
@DanielSun11 CI已完成请review

**Comment by DanielSun11** [comment]
目前paddleAPITest支持大部分api和torch的对比，请拉取最新的paddleAPITest并使用-- accuracy=True来和torch进行对比。
除了不支持对比的api，paddle_only = True的回测结果已经没有太大的参考价值。特别是需要填充默认值的情况下，需要torch对比一下，以防止后续回测继续报错。现有的测试中不该再出现paddle error，paddle_only pass的case也需要和torch对比一下

**Comment by co63oc** [comment]
> 目前paddleAPITest支持大部分api和torch的对比，请拉取最新的paddleAPITest并使用-- accuracy=True来和torch进行对比。 除了不支持对比的api，paddle_only = True的回测结果已经没有太大的参考价值。特别是需要填充默认值的情况下，需要torch对比一下，以防止后续回测继续报错。现有的测试中不该再出现paddle error，paddle_only pass的case也需要和torch对比一下

已复测更新了描述

**Comment by DanielSun11** [comment]
paddle.nn.functional.kl_div 中CPU[not compare grad]是因为反向返回了一个None，这是存在问题的，因为GPU中有反向cpu中应该也有反向，不应为None。

**Comment by co63oc** [comment]
> paddle.nn.functional.kl_div 中CPU[not compare grad]是因为反向返回了一个None，这是存在问题的，因为GPU中有反向cpu中应该也有反向，不应为None。

有两个返回值，一个为None，测试需要在PaddleAPITest增加配置
https://github.com/PFCCLab/PaddleAPITest/pull/268

**Review by DanielSun11 — APPROVED** [review]
LGTM

**Review by wanghuancoder — APPROVED** [review]
